integrations/opennms-jasper-extensions/src/main/java/org/opennms/netmgt/jasper/grafana/ExceptionToPngRenderer.java

```diff
@@ -0,0 +1,132 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2019 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2019 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.jasper.grafana;
+
+import java.awt.Color;
+import java.awt.Font;
+import java.awt.FontMetrics;
+import java.awt.Graphics2D;
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+
+import javax.imageio.ImageIO;
+
+import com.google.common.base.Throwables;
+
+/**
+ * Utility class to rendering an exception as an image.
+ *
+ * @author jwhite
+ */
+public class ExceptionToPngRenderer {
+
+    /**
+     * Use the system default font.
+     */
+    private static final Font font = new Font(null, Font.PLAIN, 48);
+
+    /**
+     * Render the given exception to an image.
+     *
+     * @param e exception
+     * @return a byte array of a .png representing the given exception
+     * @throws IOException if an error occurs rendering the .png
+     */
+    public static byte[] renderExceptionToPng(Exception e) {
+        // Build a string representation of the exception
+        final StringBuilder sb = new StringBuilder();
+        sb.append("Exception occurred: ");
+        final String stack = Throwables.getStackTraceAsString(e) ;
+        // Limit the length of the stack
+        sb.append(getFirstNLines(stack, 5));
```

Comment:
Is this something we want to make configurable via a system property?

Reply:
Good call. Done.